src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/XmlDescriptor.php

```diff
@@ -392,6 +400,41 @@ private function getContainerDefinitionDocument(Definition $definition, $id = nu
     }
 
     /**
+     * @param array $arguments
+     *
+     * @return \DOMNode[]
+     */
+    private function getArgumentNodes(array $arguments, \DOMDocument $dom)
+    {
+        $nodes = array();
+
+        foreach ($arguments as $argumentKey => $argument) {
+            $argumentXML = $dom->createElement('argument');
+
+            if (is_string($argumentKey)) {
+                $argumentXML->setAttribute('key', $argumentKey);
+            }
+
+            if ($argument instanceof Reference) {
+                $argumentXML->setAttribute('type', 'service');
+                $argumentXML->setAttribute('id', (string) $argument);
+            } elseif (is_array($argument)) {
+                $argumentXML->setAttribute('type', 'collection');
+
+                foreach ($this->getArgumentNodes($argument, $dom) as $childArgumenXML) {
+                    $argumentXML->appendChild($childArgumenXML);
+                }
+            } else {
```

Comment:
As pointed out, there should be something like:

```
} elseif ($argument instanceof Definition) {
    $argumentXML->setAttribute('type', 'service');
}
```

We don't know about its id but at least we know that the service expects it.